internal/util/BasicProviders.go

```diff
@@ -43,20 +45,99 @@ type LogConfig struct {
 	DevMode bool `env:"LOGGER_DEV_MODE" envDefault:"false"`
 }
 
+type HideSensitiveFieldsEncoder struct {
+	zapcore.Encoder
+	cfg zapcore.EncoderConfig
+}
+
+func redactField(ref *reflect.Value, i int) {
+	refField := ref.Field(i)
+	newValue := reflect.New(refField.Type()).Elem()
+	fieldType := ref.Field(i).Type().Kind()
+	// more cases can be added if required
+	switch fieldType {
+	case reflect.String:
+		newValue.SetString("[REDACTED]")
+	}
+	ref.Field(i).Set(newValue)
+}
+
+func hideSensitiveData(v interface{}) interface{} {
+	if v == nil {
+		return nil
+	}
+	ptrRef := reflect.ValueOf(v)
+	if ptrRef.Kind() != reflect.Ptr {
+		ptrRef = reflect.New(reflect.TypeOf(v))
+		ptrRef.Elem().Set(reflect.ValueOf(v))
```

Comment:
already getting this while iterating in EncodeEntry. should remove this as well as should not repeat computations again

Reply:
removed all unnecessary re-computations